### DISCUSSION TITLE
Problem in "merging" different data using add_mesh + opacity

### DISCUSSION THREAD

**Discussion by enricofacca** [pyvista/pyvista#5355] [Q&A] [answered]
Hello everyone
I am trying to plot two scalars ("tdens" and "network") from a vtu file.
I was able to plot them separately, but I would like to have the network in the background with the tdens variable on top. Similarly to the image attached.

I tried to play with different opacity level, the order of the plots but I did not get the expected results.  

```python
import pyvista as pv

path_vtu="squares.vtu"
vtu = pv.read(path_vtu)

# plot network
pl = pv.Plotter()
pl.add_mesh(vtu,
            scalars='network', 
            cmap="Greys_r",
            show_scalar_bar=False)
            
pl.view_xy()
pl.save_graphic("network.pdf")  
pl = pv.Plotter()
pl.add_mesh(vtu,
            scalars='tdens', 
            cmap="hot_r",
            show_scalar_bar=False)
pl.view_xy()
pl.save_graphic("tdens.pdf")

# Trying to merge

pl = pv.Plotter()
pl.add_mesh(vtu,
            scalars='network', 
            cmap="Greys",
            opacity=[0,0.1],
            show_scalar_bar=False)
pl.add_mesh(vtu,
            scalars='tdens', 
            cmap="hot_r",
            opacity=0.5,
            show_scalar_bar=False)
pl.view_xy()
pl.save_graphic("merge.pdf") 
```

Attached the input vtu file.

![squares_1e-5](https://github.com/pyvista/pyvista/assets/46008578/57b16d86-5d6d-4679-ada9-06d4f3d90490)
[vtu_data.zip](https://github.com/pyvista/pyvista/files/13691703/vtu_data.zip)

**Comment by enricofacca** [comment]
I realized I forgot to include the code I wrote so far...
I edited my original post.

**Reply by MatthewFlamm** [reply, marked as answer]
Is the problem that you are only getting one type of scalar data plotted?.  If so, try using `pl.add_mesh(vtu.copy(), ...` the second time.

**Reply by enricofacca** [reply]
Thank Matthew. I was so simple...
Can you also explain me why the following code produces large grey areas (output attached)? 
Is it the background? 

I would like that the cells having the "tdens" scalar below the lower bound of clim to be plotted with the "leftmost" color of the colormap. 

Thanks again for the support.

Enrico
[merge.pdf](https://github.com/pyvista/pyvista/files/13752825/merge.pdf)


```
import pyvista as pv

path_vtu="squares.vtu"
vtu = pv.read(path_vtu)

pl = pv.Plotter()
pl.add_mesh(vtu,
           scalars='network', 
            cmap="binary",
            opacity = [0,1],
            show_scalar_bar=False)
pl.add_mesh(vtu.copy(),
            scalars='tdens', 
            cmap="hot_r",
            opacity=[0.99],
            clim = [1e-4, 1e-2],
            show_scalar_bar=True)
pl.view_xy()
pl.save_graphic("merge.pdf")  
``
